.gitignore

```diff
@@ -92,6 +92,8 @@ HistomicsTKDocker.code-workspace
 histomicstk/Debugging
 .devcontainer/
 MANIFEST.in
+.pre-commit-config.yaml
+run_tests_code_cleanup.sh
```

Comment:
I merged a change which doesn't require explicit clean up between runs of `tox -e docs` (though it still is necessary when testing different versions of python because scikit-build leaves artifacts).  You might not need your local script any more.  (You can leave this here, though).

Reply:
That's great, I wrote this bash script to do other things too. Otherwise, I'll have to go around and find the exact line of code to run. It does testing, cleaning, deployment, and autopep8 to a specific location if the tox -e format doesn't work.
`#!/bin/bash
set +e

# Prompt the user to enter the first string
read -p "Select the type a-all, c-clean, d-deploy, p-autopep8, t-test: " VAR

# Function: Run tests
run_tests() {
  echo "------------> Running nuclei detection test"
  tox -e py310 -- -k test_nuclei_segmentation
  echo "------------> Running cli results test"
  tox -e py310 -- -k test_cli_results
  echo "------------> All Good"
}

# Function: Run cleanup
run_cleanup() {
  echo "------------> clean the code for erros"
  tox -e format
  echo "------------> Run Flake code checking"
  tox -e flake8
  echo "------------> Run pre-commit code check"
  pre-commit run --all-files
  echo "------------> All Good"
}

# Function: Run deployment
run_deploy() {
  echo "------------> Deploying docker"
  docker build -f Dockerfile -t sek_histomicstk:2.0 .
  echo "------------> All Good"
}

# Function: Run deployment
run_autopep8() {
  read -p "provide relative path to file: " VAR2
  autopep8 "$VAR2" --in-place --pep8-passes 2000 --verbose
  echo "------------> All Good"
}

# Function: Run everything
run_everything() {
  run_tests
  run_cleanup
  run_deploy
}

if [ "$VAR" == "t" ]; then
  run_tests
fi

if [ "$VAR" == "c" ]; then
  run_cleanup
fi

if [ "$VAR" == "d" ]; then
  run_deploy
fi

if [ "$VAR" == "p" ]; then
  run_autopep8
fi

if [ "$VAR" == "a" ]; then
  run_everything
fi
`

Reply:
I'm also a big fan of having a local bash script to manage my projects -- I usually have them outside of the repo directory and in my path.

Reply:
Okay, I'll keep it outside the main repo then.

Reply:
You don't have to -- adding the line to the .gitignore is fine.